src/backoff.rs

```diff
@@ -6,7 +6,7 @@ use tracing::info;
 /// Exponential backoff with jitter
 ///
 /// See <https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/>
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Copy)]
```

Comment:
I'm not sure if this creates any other form of violation, however, I had to switch to using `Arc` for the `BackoffConfig` just for simplicity and also to avoid having to introduce complexity with using lifetimes.